test/integration/utility.h

```diff
@@ -22,7 +23,10 @@ class BufferingStreamDecoder : public Http::StreamDecoder, public Http::StreamCa
   BufferingStreamDecoder(std::function<void()> on_complete_cb) : on_complete_cb_(on_complete_cb) {}
 
   bool complete() { return complete_; }
-  const Http::HeaderMap& headers() { return *headers_; }
+  const Http::HeaderMap& headers() {
+    ASSERT(headers_ != nullptr);
```

Comment:
STR_EQ comparisons in the integration tests like this one : https://github.com/lyft/envoy/blob/master/test/integration/server.cc#L60 cause a segfault if response->complete() is false. Would it make more sense to ASSERT_TRUE(response->complete()) and remove this change? The only problem is that this particular example is in the destructor of IntegrationTestServer.